component/core/src/test/java/org/exoplatform/social/core/jpa/storage/ProfileSettingStorageTest.java

```diff
@@ -0,0 +1,142 @@
+    try {
+      ProfilePropertySetting profilePropertySetting = createProfileSettingInstance("test");
+      profilePropertySetting = profileSettingStorage.saveProfilePropertySetting(profilePropertySetting, true);
+      profilePropertySetting = profileSettingStorage.findProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertNotNull(profilePropertySetting.getId());
+      profileSettingStorage.deleteProfilePropertySetting(profilePropertySetting.getId());
+      profilePropertySetting = profileSettingStorage.findProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertNull(profilePropertySetting);
+    } catch (Exception e) {
+      LOG.error(e.getMessage());
+      fail(e.getMessage());
+    }
+  }
+
+  public void testupdateProfilePropertySetting() {
+    try {
+      ProfilePropertySetting profilePropertySetting = createProfileSettingInstance("test");
+      profilePropertySetting = profileSettingStorage.saveProfilePropertySetting(profilePropertySetting, true);
+      profilePropertySetting = profileSettingStorage.findProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertTrue(profilePropertySetting.isActive());
+      profilePropertySetting.setActive(false);
+      profilePropertySetting = profileSettingStorage.saveProfilePropertySetting(profilePropertySetting, false);
+      profilePropertySetting = profileSettingStorage.findProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertFalse(profilePropertySetting.isActive());
+    } catch (Exception e) {
+      LOG.error(e.getMessage());
+      fail(e.getMessage());
+    }
+  }
+
+  public void testGetProfilePropertySettings() {
+    try {
+      assertEquals(0, profileSettingStorage.getPropertySettings().size());
+      ProfilePropertySetting profilePropertySetting1 = createProfileSettingInstance("test10");
+      ProfilePropertySetting profilePropertySetting2 = createProfileSettingInstance("test20");
+      ProfilePropertySetting profilePropertySetting3 = createProfileSettingInstance("test30");
+      ProfilePropertySetting profilePropertySetting4 = createProfileSettingInstance("test40");
+      ProfilePropertySetting profilePropertySetting5 = createProfileSettingInstance("test50");
+      ProfilePropertySetting profilePropertySetting6 = createProfileSettingInstance("test60");
+      profileSettingStorage.saveProfilePropertySetting(profilePropertySetting1, true);
+      profileSettingStorage.saveProfilePropertySetting(profilePropertySetting2, true);
+      profileSettingStorage.saveProfilePropertySetting(profilePropertySetting3, true);
+      profileSettingStorage.saveProfilePropertySetting(profilePropertySetting4, true);
+      profileSettingStorage.saveProfilePropertySetting(profilePropertySetting5, true);
+      profileSettingStorage.saveProfilePropertySetting(profilePropertySetting6, true);
+      assertEquals(6, profileSettingStorage.getPropertySettings().size());
```

Comment:
please test on order of retrieving properties.

Reply:
Done